### PR TITLE
IBX-98: As an Editor, I want to embed content into richtext

### PR DESCRIPTION
**Pull request by dew326**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-98](https://jira.ez.no/browse/IBX-98)
| **Bug/Improvement**| no
| **New feature**    | yes
| **Target version** | master
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Ask for Code Review.

### CODE REVIEW

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                view: (modelElement, { writer: downcastWriter }) => {
+                    const container = downcastWriter.createContainerElement('span', {
+                        'data-ezelement': 'ezembedinline',
+                        'data-ezview': 'embed-inline',
+                        class: 'ez-embed-inline',
+                    });
+                    const preview = downcastWriter.createUIElement('span', { class: 'ez-embed-content' }, function(domDocument) {
+                        const domElement = this.toDomElement(domDocument);
+
+                        domElement.innerHTML = `<svg class="ez-icon ez-icon--medium ez-icon--secondary">
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js:39]: bad indentation

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            })
+            .add((dispatcher) =>
+                dispatcher.on('attribute:contentName', (event, data, conversionApi) => {
+                    const downcastWriter = conversionApi.writer;
+                    const modelElement = data.item;
+                    const viewElement = conversionApi.mapper.toViewElement(modelElement);
+                    const preview = downcastWriter.createUIElement('span', { class: 'ez-embed-content' }, function(domDocument) {
+                        const domElement = this.toDomElement(domDocument);
+
+                        domElement.innerHTML = `<svg class="ez-icon ez-icon--medium ez-icon--secondary">
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js:60]: bad indentation

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+                    return toWidget(container, downcastWriter);
+                },
+            })
+            .add((dispatcher) =>
+                dispatcher.on('attribute:contentName', (event, data, conversionApi) => {
+                    const downcastWriter = conversionApi.writer;
+                    const modelElement = data.item;
+                    const viewElement = conversionApi.mapper.toViewElement(modelElement);
+                    const preview = downcastWriter.createUIElement('span', { class: 'ez-embed-content' }, function(domDocument) {
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js:57]: maybe fat arrowy?

File: src/bundle/Resources/public/js/CKEditor/embed/content/embed-editing.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                view: (modelElement, { writer: downcastWriter }) => {
+                    const container = downcastWriter.createContainerElement('div', {
+                        'data-ezelement': 'ezembed',
+                        'data-ezview': 'embed',
+                        class: 'ez-embed',
+                    });
+                    const preview = downcastWriter.createUIElement('p', { class: 'ez-embed-content' }, function(domDocument) {
+                        const domElement = this.toDomElement(domDocument);
+
+                        domElement.innerHTML = `<svg class="ez-icon ez-icon--medium ez-icon--secondary">
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/embed/content/embed-editing.js:38]: bad indentation

File: src/bundle/Resources/public/js/CKEditor/embed/content/embed-editing.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            })
+            .add((dispatcher) =>
+                dispatcher.on('attribute:contentName', (event, data, conversionApi) => {
+                    const downcastWriter = conversionApi.writer;
+                    const modelElement = data.item;
+                    const viewElement = conversionApi.mapper.toViewElement(modelElement);
+                    const preview = downcastWriter.createUIElement('p', { class: 'ez-embed-content' }, function(domDocument) {
+                        const domElement = this.toDomElement(domDocument);
+
+                        domElement.innerHTML = `<svg class="ez-icon ez-icon--medium ez-icon--secondary">
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/embed/content/embed-editing.js:59]: bad indentation

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+                        'X-Siteaccess': siteaccess,
+                        'X-CSRF-Token': token,
+                    },
+                    body,
+                    mode: 'same-origin',
+                    credentials: 'same-origin',
+                });
+
+                fetch(request)
+                    .then((response) => response.json())
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js:130]:
```suggestion
                    .then(eZ.helpers.request.getJsonFromResponse)
```

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-ui.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import EmbedBaseUI from '../embed-base-ui';
+
+class EmbedContentInlineUI extends EmbedBaseUI {
+    constructor(props) {
+        super(props);
+
+        this.configName = 'richtext_embed';
+        this.commandName = 'insertEmbedInline';
+        this.buttonLabel = this.editor.t('Embed Inlnie');
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-ui.js:9]: mispell I think? `Embed Inline`

File: src/bundle/Resources/public/js/CKEditor/embed/embed-base-ui.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        this.componentName = '';
+    }
+
+    getCommandOptions() {}
+
+    chooseContent() {
+        const languageCode = document.querySelector('meta[name="LanguageCode"]').content;
+        const config = JSON.parse(document.querySelector(`[data-udw-config-name="${this.configName}"]`).dataset.udwConfig);
+        const selectContent = window.eZ.richText.alloyEditor.callbacks.selectContent;
+        const mergedConfig = Object.assign(
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/embed/embed-base-ui.js:24]: Why not destruct?

File: src/bundle/Resources/public/scss/ckeditor/_general.scss [hunk 9]
```diff
@@ -0,0 +1,127 @@
+.ez-data-source__richtext {
+    background-color: $ez-white;
+    min-height: calculateRem(100px);
+    border: calculateRem(1px) solid rgba(0, 0, 0, 0.15);
+    border-radius: calculateRem(4px);
```
Comment (lucasOsti) [inline src/bundle/Resources/public/scss/ckeditor/_general.scss:5]: `$ibexa-border-radius`?

File: src/bundle/Resources/public/scss/ckeditor/_general.scss [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+    [data-ezelement='ezconfig'] {
+        display: none;
+    }
+
+    [data-ezelement='ezembed'],
+    [data-ezelement='ezembedinline'] {
+        background: $ez-embed-color-ground;
+        border: calculateRem(1px) solid $ez-embed-color-border;
+        border-radius: calculateRem(4px);
```
Comment (lucasOsti) [inline src/bundle/Resources/public/scss/ckeditor/_general.scss:49]: `$ibexa-border-radius`?

**Review by lucasOsti — CHANGES_REQUESTED** [review]

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js (hunk 3 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js:57]: cannot be an arrow function, the `this` matters.

File: src/bundle/Resources/public/scss/ckeditor/_general.scss (hunk 9 above)
Comment (dew326) [reply src/bundle/Resources/public/scss/ckeditor/_general.scss:5]: richtext should not rely on admin-ui variables

**Review by lucasOsti — APPROVED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Pull request merged**